addons/currency_rate_live/models/currency_rate_update.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+import json
+import urllib
+import xml.etree.ElementTree as ET
+
+from openerp.addons.web.controllers.main import xml2json_from_elementtree
+from openerp import api, fields, models
+
+
+class CurrencyRateProvider(models.Model):
+    _name = "currency.rate.provider"
+    _description = "Currency Rate Provider"
+
+    currency_ids = fields.Many2many('res.currency', string="Currencies to Update", required=True)
+    company_id = fields.Many2one('res.company', string="Company")
+    base_currency_id = fields.Many2one('res.currency', related='company_id.currency_id', store=True, string="Base Currency", required=True, readonly=True, default=1)
+    server_action_id = fields.Many2one('ir.actions.server', string='Name', domain=[('model_id', '=', 'currency.rate.provider')], required=True)
+
+    @api.multi
+    def process_currency_rate_data(self, request_url, response_type, preprocess):
+        self.ensure_one()
+        Currency = self.env['res.currency']
+        CurrencyRate = self.env['res.currency.rate']
+        parse_url = urllib.urlopen(request_url).read()
```

Comment:
* wrong variable name, .read() return data so it's not parse_url anymore :)
* we can use python [requests](http://docs.python-requests.org/en/latest/) for fetch data. it has clean API than urllib. in this case we can handle json response directly using r.json()